packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx

```diff
@@ -285,6 +295,50 @@ export let DataSpreadsheet = React.forwardRef(
       [cellEditorValue, onDataUpdate]
     );
 
+    useEffect(() => {
+      const currentHeaders: Array<any> = [];
+      if (Object.keys(currentColumns).length > 0) {
+        Object.keys(currentColumns).forEach((itemIndex) => {
+          if (currentColumns[itemIndex].Header) {
+            currentHeaders.push(currentColumns[itemIndex].Header);
+          }
+        });
+      }
+
+      if (previousState.selectedHeaderReorderActive) {
+        setPastColumns(currentHeaders);
+      }
+
+      if (
+        !previousState.selectedHeaderReorderActive &&
+        pastColumns.length > 0 &&
+        !headerCellHoldActive &&
+        JSON.stringify(currentHeaders) !== JSON.stringify(pastColumns)
+      ) {
+        if (currentHeaders.length > 0) {
```

Comment:
Hey, can you check if we can modify line 318 to 330 as below block ?
```
 onColDrag({
    headers: currentHeaders,
    data: activeCellContent.props.data,
  });
  if (currentHeaders.length === 0){
    setPastColumns([]);
  }
```

Reply:
Sure, checking

Reply:
This works, thanks, just pushed the code change